src/Elmish/Opaque.purs

```diff
@@ -47,33 +47,35 @@ import Unsafe.Coerce (unsafeCoerce)
 -- |      `Generic` or to provide type name, etc.), and without losing some
 -- |      performance.
 -- |   3) If such corruption proves to be a problem in the future, we can always
--- |      fall back to encoding/decoding `Json`, and pay some performance for it.
+-- |      fall back to encoding/decoding `Json`, and pay some performance for
+-- |      it.
 -- |
-newtype Ref (name :: Symbol) a = Ref (M.Object a)
+newtype Opaque (name :: Symbol) a = Opaque (M.Object a)
 
--- | Creates an instance of `Ref`. See comments on it above.
-ref :: ∀ name a. IsSymbol name => a -> Ref name a
-ref a = Ref $ M.singleton (refName (Proxy :: _ name)) a
+-- | Creates an instance of `Opaque`. See comments on it above.
+wrap :: ∀ name a. IsSymbol name => a -> Opaque name a
+wrap a = Opaque $ M.singleton (refName (Proxy :: _ name)) a
 
--- | Deconstructs an instance of `Ref`. See comments on it above.
-deref :: ∀ name a. IsSymbol name => Ref name a -> a
-deref (Ref m) =
-    -- This use of `fromJust` is justified, because the `Ref` constructor is not exported,
-    -- and the only two places where values of this type are constructed (`ref` above and
-    -- `CanReceiveFromJavaScript` below) guarantee that this key will be present.
+-- | Deconstructs an instance of `Opaque`. See comments on it above.
+unwrap :: ∀ name a. IsSymbol name => Opaque name a -> a
+unwrap (Opaque m) =
+    -- This use of `fromJust` is justified, because the `Opaque` constructor is
+    -- not exported, and the only two places where values of this type are
+    -- constructed (`wrap` above and `CanReceiveFromJavaScript` below) guarantee
+    -- that this key will be present.
```

Comment:
Couldn’t think of great names, so these are now meant to be used qualified (`Opaque.wrap`).